compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/KotlinPoetUtils.kt

```diff
@@ -185,9 +185,22 @@ public fun KtTypeReference.requireTypeName(
 
 @ExperimentalAnvilApi
 public fun KotlinType.asTypeName(): TypeName {
+  return asTypeName { true }!!
+}
+
+/**
+ * @param rawTypeFilter an optional raw type filter to allow for
+ *                      short-circuiting this before attempting to
+ *                      resolve type arguments.
+ */
+@ExperimentalAnvilApi
+public fun KotlinType.asTypeName(rawTypeFilter: (ClassName) -> Boolean = { true }): TypeName? {
```

Comment:
Hm, what's the use case for that? I also don't really like how the method is overloaded and returns a nullable value here and the other is non-nullable. That seems inconsistent.

Reply:
The use-case is to avoid type resolution of all type parameters if the raw type isn't desired. Would maybe calling this `findTypeName` be better?

Reply:
That's a Gradle idiom. `asTypeNameOrNull` would be better for Kotlin.

Reply:
That works for me!

Reply:
5235f09